src/django_components/util/component_highlight.py

```diff
@@ -0,0 +1,44 @@
+from typing import Literal, NamedTuple
+
+
+class HighlightColor(NamedTuple):
+    text_color: str
+    border_color: str
+
+
+COLORS = {
+    "component": HighlightColor(text_color="#2f14bb", border_color="blue"),
+    "slot": HighlightColor(text_color="#bb1414", border_color="red"),
+}
+
+
+def apply_component_highlight(type: Literal["component", "slot"], output: str, name: str) -> str:
+    """
+    Wrap HTML (string) in a div with a border and a highlight color.
+
+    This is part of the component / slot highlighting feature. User can toggle on
+    to see the component / slot boundaries.
+    """
+    color = COLORS[type]
+
+    output = f"""
+        <div style="
```

Comment:
There could be potentially more clever ways to achieve the highlight. The issue with wrapping the rendered content in `<div>` as I did below is that it breaks any layouting styling like flex, `position: absolute`, etc.

But since this is only for debugging, I think this is fine for now.